### PR TITLE
#125 supplier

### PR DESCRIPTION
**Pull request by sabrien-saber**

### CODE REVIEW

File: retail-managment-system/rms-ui/angular.json [hunk 1]
```diff
@@ -1,5 +1,8 @@
 {
   "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
+  "cli": {
+    "analytics": false
+  },
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/angular.json:5]: why push this?

File: retail-managment-system/rms-ui/src/app/domain/supplier/model/supplier.model.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export interface Supplier {
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/domain/supplier/model/supplier.model.ts:1]: we did folder models not model.
to continue with same structure and naming

File: retail-managment-system/rms-ui/src/app/domain/supplier/supplier.repository.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  constructor(httpClient: HttpClient) {
+    super(httpClient);
+  }
+  getResourceUrl(): string {
+    return 'supplier';
+  }
+  toServerModel(entity: Supplier) {
+    return {
+      id: entity.id,
+      version: entity.version,
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/domain/supplier/supplier.repository.ts:18]: don't send version

File: retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.html [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+              </button>
+            </td>
+          </ng-container>
+         <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
+        <tr mat-row *matRowDef="let row; columns: displayedColumns"></tr>
+        <app-loading-spinner></app-loading-spinner>
+      </table>
+      <mat-paginator [length]="pagination?.itemCount"
+      [pageSizeOptions]="[10,5,100]"
+      (page)="$event;onPaginationChange($event)"
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.html:97]: are you use $event;  ?

File: retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.scss [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+td {
+  line-height: 2;
+}
+
+.mat-column-id {
+  min-width: 60px;
+}
+// .mat-column-version {
+//   min-width: 90px;
+// }
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.scss:36]: don't push comment.

File: retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.scss [hunk 6]
```diff
@@ -0,0 +1,53 @@
+.content {
+  height: 600px;
+}
+.btn {
+  background-color: #002d40;
+  color: white;
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.scss:6]: if it exist in bootstrap why write style sheet

File: retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.ts [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+    'address',
+    'actions'
+  ];
+
+  constructor(private supplierRepository: SupplierRepository,
+    private fBuilder: FormBuilder,
+    private snackBar: MatSnackBar,
+    private dialog: MatDialog,
+    private translate: TranslateService) {
+      // this.dataSource.paginator = this.paginator;
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.ts:44]: don't push comment

File: retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.ts [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+      this.supplierForm = this.fBuilder.group({
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.ts:45]: write supplierForm in separate function to achieve single responsibility

File: retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.ts [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+    this.currentData = supplier;
+  }
+
+  updateSupplier() {
+    this.isButtonVisible = true;
+    this.submit = true;
+    this.supplierRepository.update(this.supplierForm.value).subscribe(
+      () => {
+        this.getSuppliers();
+        console.log("update");
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.ts:86]: don't push console

**Review by ZinabHashem — CHANGES_REQUESTED** [review]

File: retail-managment-system/rms-ui/angular.json (hunk 1 above)
Comment (sabrien-saber) [reply retail-managment-system/rms-ui/angular.json:5]: this is with node module

File: retail-managment-system/rms-ui/src/app/domain/supplier/model/supplier.model.ts (hunk 2 above)
Comment (sabrien-saber) [reply retail-managment-system/rms-ui/src/app/domain/supplier/model/supplier.model.ts:1]: ok

File: retail-managment-system/rms-ui/src/app/domain/supplier/supplier.repository.ts (hunk 3 above)
Comment (sabrien-saber) [reply retail-managment-system/rms-ui/src/app/domain/supplier/supplier.repository.ts:18]: modified it

File: retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.html (hunk 4 above)
Comment (sabrien-saber) [reply retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.html:97]: no deleted it

File: retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.scss (hunk 5 above)
Comment (sabrien-saber) [reply retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.scss:36]: ok

File: retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.scss (hunk 6 above)
Comment (sabrien-saber) [reply retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.scss:6]: for specific style

**Comment by ZinabHashem** [comment]
why do scroll bar with different design ?

File: retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.ts [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+    });
+  }
+
+  deleteSupplier(supplier: Supplier) {
+
+    this.supplierRepository.delete(supplier.id).subscribe(() => {
+      this.getSuppliers();
+      this.snackBar.open(
+        this.translate.instant('employee.delete-successfuly'),
+        this.translate.instant('employee.close'),
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.ts:158]: supplier not employee

File: retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.ts [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+  addSupplier() {
+    this.isButtonVisible = true;
+    this.submit = true;
+    this.supplierRepository.add(this.supplierForm.value).subscribe(
+      () => {
+        this.getSuppliers();
+        this.submit = false;
+        this.snackBar.open(
+          this.translate.instant('supplier.added-successfuly'),
+          this.translate.instant('supplier.close'),
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.ts:112]: is this translate work?

**Review by ZinabHashem — CHANGES_REQUESTED** [review]

File: retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.ts (hunk 10 above)
Comment (sabrien-saber) [reply retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.ts:157]: done

File: retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.ts (hunk 11 above)
Comment (sabrien-saber) [reply retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.ts:99]: fix it and done now

File: retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.scss (hunk 6 above)
Comment (ZinabHashem) [reply retail-managment-system/rms-ui/src/app/pages/supplier/supplier.component.scss:6]: if you are used bootstrap, it will not work?

**Pull request merged**